src/asynk/async_worker.rs

```diff
@@ -107,3 +108,70 @@ where
         }
     }
 }
+
+#[cfg(test)]
+mod async_worker_tests {
+    use super::AsyncWorker;
+    use crate::asynk::async_queue::AsyncQueueTest;
+    use crate::asynk::async_queue::AsyncQueueable;
+    use crate::asynk::async_queue::FangTaskState;
+    use crate::asynk::AsyncRunnable;
+    use crate::asynk::Error;
+    use crate::RetentionMode;
+    //use crate::SleepParams;
+    use async_trait::async_trait;
+    use bb8_postgres::bb8::Pool;
+    use bb8_postgres::tokio_postgres::NoTls;
+    use bb8_postgres::PostgresConnectionManager;
+    use serde::{Deserialize, Serialize};
+
+    #[derive(Serialize, Deserialize)]
+    struct WorkerAsyncTask {
+        pub number: u16,
+    }
+
+    #[typetag::serde]
+    #[async_trait(?Send)]
+    impl AsyncRunnable for WorkerAsyncTask {
+        async fn run(&self, _queueable: &mut dyn AsyncQueueable) -> Result<(), Error> {
+            Ok(())
+        }
+    }
+
+    #[tokio::test]
+    async fn execute_and_finishes_task() {
+        let pool = pool().await;
+        let mut connection = pool.get().await.unwrap();
+        let transaction = connection.transaction().await.unwrap();
+
+        let mut test = AsyncQueueTest {
+            transaction: Some(transaction),
+        };
+
+        let task = WorkerAsyncTask { number: 1 };
+        let metadata = serde_json::to_value(&task as &dyn AsyncRunnable).unwrap();
+
+        let task = test.insert_task(metadata, &task.task_type()).await.unwrap();
+        let id = task.id;
+
+        let mut worker = AsyncWorker::builder()
+            .queue(&mut test as &mut dyn AsyncQueueable)
+            .task_type(Some("common".to_string()))
+            .retention_mode(RetentionMode::KeepAll)
+            .build();
+        let task_finished = worker.run(task).await._get_task();
```

Comment:
as I suggested earlier the returning type will be (). 

so you'll have to fetch task by id from the db in this test to check it